### ISSUE TITLE
Error in rke-ingress-controller-deploy-job in the cluster

### ISSUE THREAD

**Issue by sowmyav27** [rancher/rancher#29745]
**What kind of request is this (question/bug/enhancement/feature request):**bug

**Steps to reproduce (least amount of steps as possible):**

- Deploy an RKE cluster - DO/Custom
- There is an error in `rke-ingress-controller-deploy-job` in the cluster. And ingress controller is not deployed. Logs of this pod:
```
namespace/ingress-nginx unchanged
configmap/nginx-configuration configured
configmap/tcp-services unchanged
configmap/udp-services unchanged
serviceaccount/nginx-ingress-serviceaccount unchanged
clusterrole.rbac.authorization.k8s.io/nginx-ingress-clusterrole unchanged
role.rbac.authorization.k8s.io/nginx-ingress-role unchanged
rolebinding.rbac.authorization.k8s.io/nginx-ingress-role-nisa-binding unchanged
clusterrolebinding.rbac.authorization.k8s.io/nginx-ingress-clusterrole-nisa-binding unchanged
error: error validating "/etc/config/rke-ingress-controller.yaml": error validating data: [ValidationError(DaemonSet.spec.template.spec.containers[0].ports[0]): missing required field "containerPort" in io.k8s.api.core.v1.ContainerPort, ValidationError(DaemonSet.spec.template.spec.containers[0].ports[1]): missing required field "containerPort" in io.k8s.api.core.v1.ContainerPort]; if you choose to ignore these errors, turn validation off with --validate=false
```

**Expected Result:**
Ingress controller should be deployed successfully

**Other details that may be helpful:**

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.4.9-rc2
- Installation option (single install/HA): Single/HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): custom/DO
- Kubernetes version (use `kubectl version`):

```
1.18.10-rancher1-2
```

**Comment by sowmyav27** [comment]
**On 2.4.9-rc4 and master-head commit id: `a3a4c8748` and 2.5-head commit id: `d87f58`**

- Deploy an RKE cluster - DO/Custom
- There is no error in rke-ingress-controller-deploy-job in the cluster.
- ingress controller is deployed successfully.
- Ingress controller version dpelloyed is `nginx-ingress-controller:nginx-0.35.0-rancher2` (1.18 on 2.4, 1.19 on master and 2.5-head)

**Comment by kinarashah** [comment]
https://github.com/rancher/rke/issues/1876

**Comment by kinarashah** [comment]
Merged for 2.5-head and master

**Comment by sowmyav27** [comment]
**On 2.5-head - commit id: `bebfe0d`**

- Deploy an RKE cluster - DO/Custom
- There is no error in rke-ingress-controller-deploy-job in the cluster.
- ingress controller is deployed successfully.
- Ingress controller version dpelloyed is nginx-ingress-controller:nginx-0.35.0-rancher2 (1.18 on 2.4, 1.19 on master and 2.5-head)
